org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ASTParser.java

```diff
@@ -245,10 +247,13 @@ private List<Classpath> getClasspath() throws IllegalStateException {
 			}
 			if (this.sourcepaths != null) {
 				for (int i = 0, max = this.sourcepaths.length; i < max; i++) {
-					String encoding = this.sourcepathsEncodings == null ? null : this.sourcepathsEncodings[i];
```

Comment:
This is a mostly-unrelated NPE check that we found was required in some instances. If you wish this to be removed from the PR I will do so.